rmw/src/discovery_params.c

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <stddef.h>
+#include <string.h>
+
+#include "rmw/discovery_params.h"
+#include "rmw/error_handling.h"
+
+rmw_discovery_params_t
+rmw_get_zero_initialized_discovery_params(void)
+{
+  rmw_discovery_params_t result = (rmw_discovery_params_t) {
+    .automatic_discovery_range = RMW_AUTOMATIC_DISCOVERY_RANGE_DEFAULT,
+    .static_peers_count = 0,
+  };  // NOLINT(readability/braces): false positive
+  return result;
+}
+
+bool
+rmw_discovery_params_equal(rmw_discovery_params_t * left, rmw_discovery_params_t * right)
+{
+  if (left->automatic_discovery_range != right->automatic_discovery_range) {
+    return false;
+  }
+
+  if (left->static_peers_count != right->static_peers_count) {
+    return false;
+  }
+
+  for (size_t ii = 0; ii < left->static_peers_count; ++ii) {
+    if (strncmp(
+        left->static_peers[ii].peer_address,
+        right->static_peers[ii].peer_address,
+        RMW_DISCOVERY_PARAMS_PEER_MAX_LENGTH) != 0)
+    {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+rmw_ret_t
+rmw_discovery_params_copy(
+  const rmw_discovery_params_t * src,
+  const rcutils_allocator_t * allocator,
+  rmw_discovery_params_t * dst)
+{
+  RMW_CHECK_ARGUMENT_FOR_NULL(src, RMW_RET_INVALID_ARGUMENT);
+  RCUTILS_CHECK_ALLOCATOR(allocator, return RMW_RET_INVALID_ARGUMENT);
+  RMW_CHECK_ARGUMENT_FOR_NULL(dst, RMW_RET_INVALID_ARGUMENT);
+
+  dst->automatic_discovery_range = src->automatic_discovery_range;
+  dst->static_peers =
+      allocator->zero_allocate(
+        src->static_peers_count, 
+        sizeof(peer_address_t),
+        allocator->state);
+  for (size_t i = 0; i < src->static_peers_count; i++)
+  {
```

Comment:
uncrustify should catch this (maybe it hasn't been run yet), but I think it should be:

```suggestion
  for (size_t i = 0; i < src->static_peers_count; i++) {
```

Reply:
Fixed: [1152b9b](https://github.com/ros2/rmw/pull/349/commits/1152b9b1392488e0b7c74ec9213d30df82f023d4)